datastore/kafka/kafka.go

```diff
@@ -16,13 +16,15 @@ import (
 
 // Producer client to handle kafka interactions
 type Producer struct {
-	kafkaProducer     *kafka.Producer
-	namespace         string
-	prometheusEnabled bool
-	metricsCollector  metrics.MetricCollector
-	logger            *logrus.Logger
-	airbrakeHandler   *airbrake.AirbrakeHandler
-	deliveryChan      chan kafka.Event
+	kafkaProducer      *kafka.Producer
+	namespace          string
+	prometheusEnabled  bool
+	metricsCollector   metrics.MetricCollector
+	logger             *logrus.Logger
+	airbrakeHandler    *airbrake.AirbrakeHandler
+	deliveryChan       chan kafka.Event
+	ackChan            chan (*telemetry.Record)
+	reliableAckTxTypes []string
```

Comment:
minor: could make this a map instead of a slice and it would improve code readability in `ProcessReliableAck`, probably no performance improvement.

Reply:
i thought about that but didn't implement it since there was no performance improvement. From readability POV, it makes sense. Updated!